codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -446,18 +450,28 @@ public static Http2Headers toHttp2Headers(HttpHeaders inHeaders, boolean validat
     public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) {
         Iterator<Entry<CharSequence, CharSequence>> iter = inHeaders.iteratorCharSequence();
         CharSequenceMap<AsciiString> connectionBlacklist =
-            toLowercaseMap(inHeaders.getAll(HttpHeaderNames.CONNECTION));
+            toLowercaseMap(inHeaders.getAll(CONNECTION));
         while (iter.hasNext()) {
             Entry<CharSequence, CharSequence> entry = iter.next();
             final AsciiString aName = AsciiString.of(entry.getKey()).toLowerCase();
             if (!HTTP_TO_HTTP2_HEADER_BLACKLIST.contains(aName) &&
                 !connectionBlacklist.contains(aName)) {
                 // https://tools.ietf.org/html/rfc7540#section-8.1.2.2 makes a special exception for TE
-                if (aName.contentEqualsIgnoreCase(HttpHeaderNames.TE)) {
-                    if (AsciiString.containsIgnoreCase(entry.getValue(), HttpHeaderValues.TRAILERS)) {
-                        out.add(HttpHeaderNames.TE, HttpHeaderValues.TRAILERS);
+                if (aName.contentEqualsIgnoreCase(TE)) {
+                    if (AsciiString.indexOf(entry.getValue(), ',', 0) == -1) {
+                        if (AsciiString.contentEqualsIgnoreCase(entry.getValue(), TRAILERS)) {
```

Comment:
should we trim this?

Reply:
good question... @carl-mastrangelo @Scottmitch @ejona86 WDYT ?

Reply:
Wouldn't hurt, but doesn't seem necessary. Although it's probably reading too much into the language, the spec does say 'it MUST NOT contain any value other than "trailers".' :)

Reply:
Agreed with @ejona86 ... I would prefer to avoid trimming unless an argument can be made that the RFC requires it.

Reply:
Maybe we should trim just to be safe:

https://tools.ietf.org/html/rfc7230#section-3.2
> header-field   = field-name ":" OWS field-value OWS

https://tools.ietf.org/html/rfc7230#section-3.2.3
>  OWS            = *( SP / HTAB )
                    ; optional whitespace